apollo-federation/src/source_aware/federated_query_graph/graph_path.rs

```diff
@@ -25,12 +25,13 @@ pub(crate) struct FederatedGraphPath {
 
 #[derive(Debug)]
 pub(crate) struct Edge {
-    operation_element: Option<Arc<OperationPathElement>>,
-    edge: Option<EdgeIndex>,
-    self_condition_resolutions_for_edge: IndexMap<SelfConditionIndex, ConditionResolutionId>,
-    source_entering_condition_resolutions_at_head:
+    pub operation_element: Option<Arc<OperationPathElement>>,
```

Comment:
It might not be correct for these to be pub.

graph paths and path trees seem quite intertwined (since a path tree is made up of graph paths and a graph path can have conditions expressed as path trees). It might make more sense for both of them to be in the same module so they can freely access each other's internals.